crates/lib/src/executable.rs

```diff
@@ -303,8 +344,8 @@ impl Executable<'_, '_> {
         self.qvm = Some(qvm);
         result
             .map_err(Error::from)
-            .map(|registers| execution_data::Qvm {
-                registers,
+            .map(|registers| execution_data::ExecutionData {
+                readout_data: ReadoutData::Qvm(registers),
```

Comment:
One thing I've learned from the past few weeks of discussions is that we should label readout data and memory accurately. This was a mistake I introduced to pyQuil v3 and so it's no surprise that it's in the Rust SDK here ... but the QVM does not have readout data. This is memory.

Where things get weird is that we _only_ really have readout data for QPU right now when it comes to reconstructing per-shot memory. So, for QPU data, there should be a method like `build_memory_matrix_from_readout_data` (or better name) which we'd deprecate when we have a better option.

Reply:
Can you help clarify your idea for `build_memory_matrix_from_readout_data` with a function signature? Trying to understand what the ask is.

The to be renamed `to_readout_map` method effectively builds memory matrices from readout data if it is possible for every register. Would `build_memory_matrix_from_readout_data` do it for a single register?